### ISSUE TITLE
support a way to download subset of github release assets

### ISSUE THREAD

**Issue by jamespollard8** [carvel-dev/vendir#7]
In cf-for-k8s, we'd like to remove `disableAutoChecksumValidation: true` from our entry for kpack, for example.

That block looks like this:
```
  - path: github.com/pivotal/kpack
    githubRelease:
      slug: pivotal/kpack
      tag: v0.0.8
      disableAutoChecksumValidation: true
    includePaths:
    - release-0.0.8.yaml
```

And when we remove the `disableAutoChecksumValidation` line, we get the error:

`Error: Syncing directory 'config/_ytt_lib': Syncing directory 'github.com/pivotal/kpack' with github release contents: Finding checksums in release notes: Expected to find sha256 checksum for file 'logs-v0.0.8-linux.tgz'`

...even though the only file in our include path is `release-0.0.8.yaml`.

Here's the kpack release page in case that's helpful: https://github.com/pivotal/kpack/releases/tag/v0.0.8

**Comment by jamespollard8** [comment]
bump

**Comment by cppforlife** [comment]
i was thinking more about this... i think a better solution here is to have checksums in release notes for all assets included (not sure why kpack folks only did it for one file). vendir today checks integrity of downloaded asset (in this case github release is considered to be "one" thing), and at a later stage it includes/excludes files. it's going to be quite unnatural to intermingle the too since files could even be rearranged (there wouldnt be any connection to original checksum).

do you think we can pursue adding checksums for all files to kpack asset?

**Comment by jamespollard8** [comment]
That's definitely something we could pursue but I worry that those changes are unlikely to get prioritized by pack folks.

I'd also be okay marking this general issue as "won't fix" - I just want to make sure we're being intentional here.

I'll defer to whichever decision you're leaning towards - just let us know.

**Comment by cppforlife** [comment]
added `assetNames` for githubRelease sync: https://github.com/k14s/vendir/commit/1bf2caa4f5fb74b3ca65e28dc8f78f8aee024c1c. will be available in v0.12.0+.

**Comment by cppforlife** [comment]
included in https://github.com/k14s/vendir/releases/tag/v0.12.0
